spec/models/agents/post_agent_spec.rb

```diff
@@ -311,8 +311,21 @@
   end
 
   describe "#working?" do
+    it "checks if there was an error" do
+      @checker.error("error")
+      expect(@checker.logs.count).to eq(1)
+      expect(@checker.reload).not_to be_working
+    end
+
+    it "checks if no event has been received" do
```

Comment:
It would be nice to test your `expected_receive_period_in_days` change here.
```
@checker.options.delete('expected_receive_period_in_days')
expect(@checker).to be_working
```

The check failing when no event was received was already part of the original `"checks if events have been received within expected receive period"` test.